include/sparrow/buffer/buffer_view.hpp

```diff
@@ -342,7 +344,14 @@ namespace sparrow
     template <class T>
     buffer_view<T>::operator buffer<std::remove_const_t<T>>() const
     {
-        return {p_data, p_data + m_size};
+        if (!p_data)
+        {
+            return buffer<std::remove_const_t<T>>();
+        }
+        else
+        {
+            return buffer<std::remove_const_t<T>>(cbegin(), cend());
```

Comment:
Why not returning this all the time? The buffer constructor should handle the case where ` cbegin() == cend()`. If not it should be fixed.

Reply:
The constructor which takes iterators does an allocation.

Reply:
Yes but since cbegin() == cend(), the allocation should be 0 byte, i.e. a noop.